test/integration/scheduler_perf/util.go

```diff
@@ -290,6 +341,10 @@ type throughputCollector struct {
 	schedulingThroughputs []float64
 	labels                map[string]string
 	namespaces            []string
+
+	progress                     []podScheduling
+	start, end                   time.Time
+	startScheduled, endScheduled int
 }
 
 func newThroughputCollector(podInformer coreinformers.PodInformer, labels map[string]string, namespaces []string) *throughputCollector {
```

Comment:
-1 to this change. I think you are mixing latency and throughput in a single function.

Reply:
"this change" = a1a92e5601cb11e9b31f3c0cab6cf1bd238f0c1b = `scheduler_perf: also record metric for "total pods/s"`?

I wouldn't say that this adds latency measurements. It's still the same metric (pods/s), just determined over the entire duration of the measurement.

In practice, it's close to the Perc99:
```
name                                                           SchedulingThroughput/Average
PerfScheduling/SchedulingBasic/500Nodes-36                                                                  1.00k ± 0%
PerfScheduling/SchedulingPodAntiAffinity/500Nodes-36                                                          200 ± 0%
PerfScheduling/SchedulingSecrets/500Nodes-36                                                                  724 ± 0%
PerfScheduling/SchedulingInTreePVs/500Nodes-36                                                                364 ± 0%
PerfScheduling/SchedulingMigratedInTreePVs/500Nodes-36                                                        363 ± 0%
PerfScheduling/SchedulingCSIPVs/500Nodes-36                                                                   350 ± 0%
PerfScheduling/SchedulingPodAffinity/500Nodes-36                                                              250 ± 0%
PerfScheduling/SchedulingPreferredPodAffinity/500Nodes-36                                                     500 ± 0%
PerfScheduling/SchedulingPreferredPodAntiAffinity/500Nodes-36                                                 500 ± 0%
PerfScheduling/SchedulingNodeAffinity/500Nodes-36                                                             537 ± 0%
PerfScheduling/TopologySpreading/500Nodes-36                                                                  500 ± 0%
PerfScheduling/PreferredTopologySpreading/500Nodes-36                                                         500 ± 0%
PerfScheduling/MixedSchedulingBasePod/500Nodes-36                                                             333 ± 0%
PerfScheduling/PreemptionPVs/500Nodes-36                                                                     45.4 ± 0%
PerfScheduling/Unschedulable/500Nodes/200InitPods-36                                                          500 ± 0%

...

name                                                           SchedulingThroughput/Perc99
PerfScheduling/SchedulingBasic/500Nodes-36                                                                  1.00k ± 0%
PerfScheduling/SchedulingPodAntiAffinity/500Nodes-36                                                          295 ± 0%
PerfScheduling/SchedulingSecrets/500Nodes-36                                                                  908 ± 0%
PerfScheduling/SchedulingInTreePVs/500Nodes-36                                                                511 ± 0%
PerfScheduling/SchedulingMigratedInTreePVs/500Nodes-36                                                        485 ± 0%
PerfScheduling/SchedulingCSIPVs/500Nodes-36                                                                   567 ± 0%
PerfScheduling/SchedulingPodAffinity/500Nodes-36                                                              346 ± 0%
PerfScheduling/SchedulingPreferredPodAffinity/500Nodes-36                                                     606 ± 0%
PerfScheduling/SchedulingPreferredPodAntiAffinity/500Nodes-36                                                 586 ± 0%
PerfScheduling/SchedulingNodeAffinity/500Nodes-36                                                             985 ± 0%
PerfScheduling/TopologySpreading/500Nodes-36                                                                  549 ± 0%
PerfScheduling/PreferredTopologySpreading/500Nodes-36                                                         616 ± 0%
PerfScheduling/MixedSchedulingBasePod/500Nodes-36                                                             532 ± 0%
PerfScheduling/PreemptionPVs/500Nodes-36                                                                      429 ± 0%
PerfScheduling/Unschedulable/500Nodes/200InitPods-36                                                          528 ± 0%
```

Note that getting exactly 1000 pod/s for PerfScheduling/SchedulingBasic/500Nodes seems to be an artifact: I haven't checked, but I bet there's exactly one poll interval in which all pods get scheduled.

One can argue both ways: it's both redundant and provides a different view... I can take out the commit if it's too controversial.

Reply:
> One can argue both ways: it's both redundant and provides a different view... I can take out the commit if it's too controversial.

Not sure if it is. This PR is so long that it's hard to tell what is required and what is a new test feature. If you can remove it for now, I'd appreciate.

I know you tried to split everything into different commits, but as long as github doesn't show incremental updates well enough, it doesn't help much.

Reply:
I'll try to strip out as many changes as possible from this PR.